### PR TITLE
Création des classes CaseDeJeu, Joueur et Partie

### PR DESCRIPTION
**Pull request by DarkNekoGreen**
CaseDeJeu permet gérer une case sur le plateau.
Partie permet de créer une partie de jeu ainsi que le plateau de jeu
Joueur permet de gérer les joueurs et leurs déplacements

### CODE REVIEW

File: laravel5/app/Monomalpoly/CaseDeJeux.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+class CaseDeJeux
```
Comment (floo51) [inline laravel5/app/Monomalpoly/CaseDeJeux.php]: Naming should be in english

File: laravel5/app/Monomalpoly/CaseDeJeux.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	private $y;
+
+	public function Case($nom,$x,$y){
+		$this->listeJoueur = [];
+		$this->nom = $nom;
+		$this->x = $x;
+		$this->y = $y;
+	}
+
+	public function getPlayersIn(){
```
Comment (floo51) [inline laravel5/app/Monomalpoly/CaseDeJeux.php]: Do you actually need a getter or would making the property public be enough?

File: laravel5/app/Monomalpoly/CaseDeJeux.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+class CaseDeJeux
+{
+	private $type;
+	private $nom;
+	private $listeJoueurs;
```
Comment (floo51) [inline laravel5/app/Monomalpoly/CaseDeJeux.php]: Should the game box really have a list of players? I thought it was the player's pawn which was on a box, not the other way around.

File: laravel5/app/Monomalpoly/Joueur.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		$this->couleur = $couleur;
+		
+	}
+
+	function move(){
+		$firstDice = random_int(1,6);
+		$secondDice = random_int(1,6);
+		$moveNb = $firstDice + $secondDice;
+		$canMove = testPosition($moveNb);
+		if($canMove == true){
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Joueur.php]: Same as `if($canMove){`

Comment (floo51) [reply laravel5/app/Monomalpoly/Joueur.php]: Rather than testing if it is normal and then doing the move, you better should test before actually doing the move.

Exemple:
```
if(!$canMove){
    throw new Exception("Some message");
}
```

This way, you forget handling edge cases much less often.

File: laravel5/app/Monomalpoly/Joueur.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
+<?php
+class Joueur{
+	private $id;
+	private $nom;
+	private $couleur;
+	private $posX;
+	private $posY;
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Joueur.php]: Is `posY` of any use?

File: laravel5/app/Monomalpoly/Joueur.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	private $nbDisque;
+	private $listeServeur;
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Joueur.php]: Does `listeServeur` belong here?

File: laravel5/app/Monomalpoly/Partie.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	}
+
+	function joueurJoin(){
+		//On créé un nouveau joueur
+		$listeJoueur.add(new Joueur($_SESSION['id'],$_SESSION['nom'],$listeCouleur[$listeJoueur.length()]));
+	}
+
+	function checkNbJoueur(){
+		if($listeJoueur.length()<2){
+			restartTimer();
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Partie.php]: Where is the function `restartTimer`? What does it do?

File: laravel5/app/Monomalpoly/Partie.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+		for($i = 0 ;$i<$listeNomCase.length(); $i++){
+			/* Systeme pour l'affichage en x/y */
+
+			$this->plateau.add(new CaseDeJeux($listeNomCase[$i],$x,$y));
+		}
+	}
+
+	function joueurJoin(){
+		//On créé un nouveau joueur
+		$listeJoueur.add(new Joueur($_SESSION['id'],$_SESSION['nom'],$listeCouleur[$listeJoueur.length()]));
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Partie.php]: This line is hardly readable. Split it into 2 lines.
Bonus: if you split it into 2 lines, you won't need the comment

File: laravel5/app/Monomalpoly/Partie.php [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+		/* Initialisation du plateau (liste de cases)*/
+		for($i = 0 ;$i<$listeNomCase.length(); $i++){
+			/* Systeme pour l'affichage en x/y */
+
+			$this->plateau.add(new CaseDeJeux($listeNomCase[$i],$x,$y));
+		}
+	}
+
+	function joueurJoin(){
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Partie.php]: Is it french or english?

File: laravel5/app/Monomalpoly/Partie.php (hunk 8 above)
Comment (floo51) [reply laravel5/app/Monomalpoly/Partie.php]: Using $_SESSION within a domain class is not a good practice because it increases the coupling between the class and the technology you're using. Pass needed information in the arguments of the function.

File: laravel5/app/Monomalpoly/Partie.php [hunk 10]
```diff
@@ -0,0 +1,34 @@
+<?php
+include('Case.php');
+
+class Partie{
+	private $listeCouleur = ['red','yellow','orange','blue','green','grey'];
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Partie.php]: These are called magic strings, you should use constants.

File: laravel5/app/Monomalpoly/Partie.php [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+	private $listeNomCase = [];
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Partie.php]: Probably bad naming. It makes me believe "game boxes" have a name and that there's a list for it.

File: laravel5/app/Monomalpoly/Joueur.php [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+		$this->listeServeur = [];
+
+		$this->id = $id;
+		$this->nom = $nom;
+		$this->couleur = $couleur;
+		
+	}
+
+	function move(){
+		$firstDice = random_int(1,6);
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Joueur.php]: Probably bad naming. Is it a dice or the result of a dice roll?

File: laravel5/app/Monomalpoly/Joueur.php [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+		$secondDice = random_int(1,6);
+		$moveNb = $firstDice + $secondDice;
+		$canMove = testPosition($moveNb);
+		if($canMove == true){
+			$this->position += $moveNb;
+		}
+		//à finir 
+	}
+
+	function testPosition($move){
```
Comment (floo51) [inline laravel5/app/Monomalpoly/Joueur.php]: It returns nothing. And probably badly named.
